lib/src/sql/value/value.rs

```diff
@@ -150,6 +151,7 @@ pub enum Value {
 	Constant(Constant),
 	// Closure(Box<Closure>),
 	Function(Box<Function>),
+	MlModel(Box<MlModel>),
```

Comment:
@tobiemh I was curious about whether adding variants in the middle of an enum requires bumping the `revision`.

When I checked `revision` source code, the variant index has the following data flow:
1. https://github.com/surrealdb/revision/blob/f6a6c6d67cdd0fae5949df9467a0be34d53621f4/derive/src/descriptors/enum_desc.rs#L38 (index of variant in enum)
2. https://github.com/surrealdb/revision/blob/f6a6c6d67cdd0fae5949df9467a0be34d53621f4/derive/src/descriptors/enum_desc.rs#L53
3. https://github.com/surrealdb/revision/blob/f6a6c6d67cdd0fae5949df9467a0be34d53621f4/derive/src/fields/enum_tuple.rs#L55
4. https://github.com/surrealdb/revision/blob/f6a6c6d67cdd0fae5949df9467a0be34d53621f4/derive/src/fields/enum_tuple.rs#L93
5. https://github.com/surrealdb/revision/blob/f6a6c6d67cdd0fae5949df9467a0be34d53621f4/derive/src/fields/enum_tuple.rs#L123 (serialized as integer)

It looks to me that new variants must always be added at the end _in addition to_ bumping the revision and annotating the start revision of the new variant.

I do realize that some variants, including `MlModel`, are part of the "uncomputed values" section which *may* be subject to different compatibility requirements. Compatibility of uncomputed values is, at the very least, relevant if queries are parsed on the client and sent (in their serialized form) to the SurrealDB server.

**I recommend developing clear guidelines for making format revisions in a way that preserves desired compatibility. A doc comment on `Value`, the most probable subject of revisions, would go a long way.**

Reply:
Hey @finnbear it should be enough to just add it at the bottom of the variants.